### PR TITLE
Active_demo and some minor bug fixes on FaceRecognition

### PR DESCRIPTION
**Pull request by Pavlos-Tosidis**
Changed Face Recognition inference return. Now class is 0/1 if not found/found
Added feature_extraction method to get access to features from a face image
Added active_demo in which features in database get updated with new features of the same person.

### CODE REVIEW

**Review by omichel — CHANGES_REQUESTED** [review]
From source tests, it seems there are some PEP8 errors, please check and fix them.

File: src/opendr/perception/face_recognition/face_recognition_learner.py [hunk 1]
```diff
@@ -439,8 +443,7 @@ def fit_reference(self, path=None, save_path=None, create_new=True):
                     avg_features = features_sum / total
```
Comment (omichel) [inline src/opendr/perception/face_recognition/face_recognition_learner.py:443]: This is causing PEP8 test to fail:
```
/home/runner/work/opendr/opendr/src/opendr/perception/face_recognition/face_recognition_learner.py:443:21: local variable 'avg_features' is assigned to but never used
```
```suggestion
```

File: src/opendr/perception/face_recognition/face_recognition_learner.py [hunk 2]
```diff
@@ -534,6 +538,67 @@ def infer(self, img):
...
+            transforms.Resize([int(128 * self.input_size[0] / 112), int(128 * self.input_size[0] / 112)]),
+            transforms.CenterCrop([self.input_size[0], self.input_size[1]]),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=self.rgb_mean, std=self.rgb_std)]
+        )
+        if self.mode == 'backbone_only':
+            distance = 100.0
+            if distance == 0:
+                distance = 10.0
+            person = None
```
Comment (omichel) [inline src/opendr/perception/face_recognition/face_recognition_learner.py:567]: This one is also causing PEP8 to fail:
```
/home/runner/work/opendr/opendr/src/opendr/perception/face_recognition/face_recognition_learner.py:567:13: local variable 'person' is assigned to but never used
```
```suggestion
```

**Review by omichel — CHANGES_REQUESTED** [review]

File: src/opendr/perception/face_recognition/face_recognition_learner.py (hunk 2 above)
Comment (Pavlos-Tosidis) [reply src/opendr/perception/face_recognition/face_recognition_learner.py:567]: We are working with @thomaspeyrucain on this one, so I will ask for a review when it is completed

**Comment by thomaspeyrucain** [comment]
Hello @Pavlos-Tosidis,

I am testing the tool and sometimes I get this error:
```
[ERROR] [1697016110.932028]: bad callback: <bound method FaceRecognitionNode.callback of <__main__.FaceRecognitionNode object at 0x7fcdc1274550>>
Traceback (most recent call last):
  File "/opt/ros/noetic/lib/python3/dist-packages/rospy/topics.py", line 750, in _invoke_callback
    cb(msg)
  File "/home/thomaspeyrucain/Desktop/OpenDR/opendr_pal_use_case/projects/opendr_ws/src/opendr_perception/scripts/face_recognition_node.py", line 180, in callback
    self.recognizer.database[result.description].append(features_to_compare)
AttributeError: 'Tensor' object has no attribute 'append'

```

**Comment by passalis** [comment]
@Pavlos-Tosidis There seems to be pep8 error still.

**Comment by passalis** [comment]
Thanks @Pavlos-Tosidis! Is this ready for review?

**Comment by Pavlos-Tosidis** [comment]
Yes this is ready to be reviewed

**Review by Pavlos-Tosidis — COMMENTED** [review]
Fixed pep8 errors

File: src/opendr/perception/face_recognition/face_recognition_learner.py [hunk 3]
```diff
@@ -534,6 +535,66 @@ def infer(self, img):
         else:
             raise UserWarning('Infer should be called either with backbone_only mode or with a classifier head')
 
+    def feature_extraction(self, img):
+        """
+        This method is used to perform face recognition on an image.
+
+        :param img: image to run inference on
+        :rtype img: engine.data.Image class object
```
Comment (tsampazk) [inline src/opendr/perception/face_recognition/face_recognition_learner.py:543]:
```suggestion
        :type img: engine.data.Image class object
```

**Review by tsampazk — CHANGES_REQUESTED** [review]
Thank you @Pavlos-Tosidis for the new features! I have left some comments.

A general note: in all modified/new files, the retinaface detector is initialized with `mnet` which if i recall correctly is used to identify masked faces. Is this intented?

**Review by passalis — APPROVED** [review]
Thank you!

**Review by tsampazk — APPROVED** [review]
Thank you!

**Pull request merged**